lib/index.js

```diff
@@ -71,6 +71,19 @@ Strategy.prototype.authenticate = function (req, options) {
   if (req.query && req.query.error) {
     return this.fail(req.query.error);
   }
+
+  if (req.query.code) {
+    // The code param is sent to the /callback route
+    // where code exchange and other verification will happen
+    this._verify = verifyWrapper(this._verify, this.options, req.session.authParams);
+  } else {
+    // Store auth params in the session in order for verification on callback
+    req.session.authParams = {};
+    req.session.authParams.scope = options.scope;
+    req.session.authParams.nonce = crypto.randomBytes(16).toString('hex');
```

Comment:
I get "crypto not defined" when I load this route so I think we need `var crypto = require('crypto');` at the top